HeartbeatLogging/Tests/Integration/HeartbeatLoggingIntegrationTests.swift

```diff
@@ -13,16 +13,297 @@
 // limitations under the License.
 
 import XCTest
-import HeartbeatLogging
+@testable import HeartbeatLogging
 
 class HeartbeatLoggingIntegrationTests: XCTestCase {
+  // 2021-11-01 @ 00:00:00 (EST)
+  let date = Date(timeIntervalSince1970: 1_635_739_200)
+
   override func setUpWithError() throws {
-    // Put setup code here. This method is called before the
-    // invocation of each test method in the class.
+    try removeUnderlyingHeartbeatStorageContainers()
   }
 
   override func tearDownWithError() throws {
-    // Put teardown code here. This method is called after the
-    // invocation of each test method in the class.
+    try removeUnderlyingHeartbeatStorageContainers()
+  }
+
+  /// This test may flake if it is executed during the transtion from one day to the next.
+  func testLogAndFlush() throws {
+    // Given
+    let heartbeatController = HeartbeatController(id: #function)
```

Comment:
This uses the shared storage, right? It may run into issues with other tests. It probably makes sense to use the initializer that passes in the custom storage that won't conflict with any other functions.

Reply:
Yes, it uses shared storage, but because shared storage is keyed by the `id` used to create the controller instance. We are safe from conflict here because we pass in a unique `id` specific to this test case: `#function`